python-checks/src/main/java/org/sonar/python/checks/StringFormatCorrectnessCheck.java

```diff
@@ -0,0 +1,234 @@
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.python.checks;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import org.sonar.check.Rule;
+import org.sonar.plugins.python.api.SubscriptionContext;
+import org.sonar.plugins.python.api.symbols.Symbol;
+import org.sonar.plugins.python.api.tree.BinaryExpression;
+import org.sonar.plugins.python.api.tree.CallExpression;
+import org.sonar.plugins.python.api.tree.DictionaryLiteral;
+import org.sonar.plugins.python.api.tree.DictionaryLiteralElement;
+import org.sonar.plugins.python.api.tree.Expression;
+import org.sonar.plugins.python.api.tree.KeyValuePair;
+import org.sonar.plugins.python.api.tree.RegularArgument;
+import org.sonar.plugins.python.api.tree.StringElement;
+import org.sonar.plugins.python.api.tree.StringLiteral;
+import org.sonar.plugins.python.api.tree.Token;
+import org.sonar.plugins.python.api.tree.Tree;
+import org.sonar.python.tree.TreeUtils;
+
+@Rule(key = "S3457")
+public class StringFormatCorrectnessCheck extends AbstractStringFormatCheck {
+
+  private static final List<String> LOGGER_METHODS = Arrays.asList("logging.debug", "logging.info", "logging.warning", "logging.error", "logging.critical");
+
+  @Override
+  public void initialize(Context context) {
+    context.registerSyntaxNodeConsumer(Tree.Kind.MODULO, this::checkPrintfStyle);
+    context.registerSyntaxNodeConsumer(Tree.Kind.CALL_EXPR, ctx -> {
+      CallExpression callExpression = (CallExpression) ctx.syntaxNode();
+
+      Symbol symbol = callExpression.calleeSymbol();
+      if (symbol == null || symbol.fullyQualifiedName() == null) {
+        return;
+      }
+
+      if (LOGGER_METHODS.contains(symbol.fullyQualifiedName())) {
+        checkLoggerLog(ctx, callExpression);
+      } else {
+        this.checkStrFormatStyle(ctx);
+      }
+    });
+    context.registerSyntaxNodeConsumer(Tree.Kind.STRING_LITERAL, StringFormatCorrectnessCheck::checkFStringLiteral);
+  }
+
+  private static void checkFStringLiteral(SubscriptionContext ctx) {
+    StringLiteral literal = (StringLiteral) ctx.syntaxNode();
+    if (TreeUtils.firstAncestorOfKind(literal, Tree.Kind.PLUS) != null) {
```

Comment:
Not sure if this is a common case, but you could probably be more restrictive in case you'd want to catch wrongly formatted f-strings within uses of `PLUS` not related to the concatenation of said f-strings, like `foo(f"hello") + bar(f"world")`.